nuxeo-services/nuxeo-platform-directory/nuxeo-platform-directory-test/src/test/java/org/nuxeo/directory/test/TestInitLoadDirectory.java

```diff
@@ -0,0 +1,156 @@
+        assertDirectoryIsInitialized();
+        // Test case NEVER - UPDATE
+        hotDeployer.deploy(
+                "org.nuxeo.ecm.directory.tests:csv-never-and-update-duplicate-directory-override-contrib.xml");
+        try (Session session = directoryService.open(CSV_LOAD_DIRECTORY)) {
+            // We verify that the 2 csv files have been correctly merged
+            DocumentModelList entries = queryFullDirectory(session);
+            assertEquals(4, entries.size());
+            assertNotNull(session.getEntry("8"));
+            assertEquals("European Union", queryEurope(session).getPropertyValue("label"));
+        }
+    }
+
+    @Test
+    @Deploy("org.nuxeo.ecm.directory.tests:csv-loaded-directory-contrib.xml")
+    public void testInitDirectoryWithOnMissingColumns() throws Exception {
+        assertDirectoryIsInitialized();
+        // First we test with dataLoadingPolicy = never_load
+        hotDeployer.deploy(
+                "org.nuxeo.ecm.directory.tests:csv-on-missing-columns-and-never-load-directory-override-contrib.xml");
+        // We verify that nothing has changed
+        assertDirectoryIsInitialized();
+
+        // Then we test with dataLoadingPolicy = ignore_duplicate
+        hotDeployer.deploy(
+                "org.nuxeo.ecm.directory.tests:csv-on-missing-columns-and-ignore-duplicate-directory-override-contrib.xml");
+
+        try (Session session = directoryService.open(CSV_LOAD_DIRECTORY)) {
+            // We verify that the 2 csv files have been correctly merged with duplicate line ignored
+            DocumentModelList entries = queryFullDirectory(session);
+            assertEquals(4, entries.size());
+            // assert unchanged existing entry:
+            assertEquals("Europe", queryEurope(session).getPropertyValue("label"));
+        }
+        // Then we test with dataLoadingPolicy = update_duplicate
+        hotDeployer.deploy(
+                "org.nuxeo.ecm.directory.tests:csv-on-missing-columns-and-update-duplicate-directory-override-contrib.xml");
+        try (Session session = directoryService.open(CSV_LOAD_DIRECTORY)) {
+            // We verify that the 2 csv files have been correctly merged
+            DocumentModelList entries = queryFullDirectory(session);
+            assertEquals(4, entries.size());
+            assertNotNull(session.getEntry("8"));
+            assertEquals("European Union", queryEurope(session).getPropertyValue("label"));
+        }
+        // Then we test with dataLoadingPolicy = error_on_duplicate
+        hotDeployer.deploy(
+                "org.nuxeo.ecm.directory.tests:csv-on-missing-columns-and-error-on-duplicate-directory-override-contrib.xml");
+        try (Session session = directoryService.open(CSV_LOAD_DIRECTORY)) {
+            // since Exception is nesed by hotDeployer we can only check that Directory is not changed
```

Comment:
I meant `hotDeployer.deploy(
                "org.nuxeo.ecm.directory.tests:csv-on-missing-columns-and-error-on-duplicate-directory-override-contrib.xml");` is supposed to throw en Exception (because of "error_on_duplicate") ; but this Exception is caught and I cannot check directly on it.